hal/src/gpio/v2/reg.rs

```diff
@@ -262,20 +260,8 @@ pub(super) unsafe trait RegisterInterface {
     }
 
     #[inline]
-    fn pincfg(&self) -> PINCFG_R {
-        let group = self.group();
-        // Safe because we only read from the registers
-        unsafe { (*group).pincfg[self.id().num as usize].read() }
-    }
-
-    #[inline]
-    fn pincfg_mut(&mut self) -> &mut PINCFG {
-        let group = self.group_mut();
-        // A mutable reference is safe here, because it points to the PINCFG
-        // register of the corresponding `PinId`. Each `Pin` is a singleton, and
-        // we have exclusive control of it through `&mut self`, so the reference
-        // is guaranteed to be unique
-        unsafe { &mut (*group).pincfg[self.id().num as usize] }
+    fn pincfg(&self) -> &PINCFG {
+        &self.group().pincfg[self.id().num as usize]
     }
```

Comment:
`pincfg_mut` was indeed unnecessary, since interior mutability is erased at the method level.